src/Server/Server.fs

```diff
@@ -510,11 +499,32 @@ let azureAccount() = task {
         return Error $"{errorType}: {error.Message}"
 }
 
+let getDefaultSubscription() = task {
+    let! account = azureAccount() |> Async.AwaitTask
+    return match account with
+           | (Ok t) ->
+               let subscriptionId = t.subscriptionId
+               let subscriptions = armClient().GetSubscriptions() |> List.ofSeq |> List.filter (fun sub -> sub.Id.SubscriptionId = subscriptionId)
+               match subscriptions |> Seq.length with
+               | 1 -> List.head(subscriptions)
+               | _ -> raise (new InvalidOperationException("Cannot get default subscription"))
```

Comment:
In the context of this application, you almost never need to actually raise an exception. The function `azureAccount()` returns `Task<Result<AzureAccount, string>>` where `Result<AzureAccount, string>` is a union:
```
AzureAccount | Error(string)
```
In the case of the `Error` you can simply propagate that back as the value you want to return, potentially adding more information to the error. 

What we can do here when we are trying to find the matching subscription is that we can still default to the subscription we get from the `armClient()` instead of erroring so that if for whatever reason shelling out to `az` doesn't work, we still get a subscription
```fs
match subscriptions |> Seq.length with
| 1 -> List.head(subscriptions)
| _ -> armClient.GetDefaultSubscription()
```

Reply:
Here is an example of how this function can be refactored, using `match!` which combines `let!` and `match`
```fs
let getDefaultSubscription() = task {
    match! azureAccount() with
    | Ok account ->
        let defaultSubscription =
            armClient().GetSubscriptions()
            |> Seq.tryFind (fun sub -> sub.Id.SubscriptionId = account.subscriptionId)
            |> Option.defaultValue (armClient().GetDefaultSubscription())

        return Ok defaultSubscription

    | Error error ->
        return Error $"Error occurred while getting the default subscription: {error}"
}
```
Although now the function returns `Task<Result<Subscription, string>>` so the callers need to handle the error case (just propagating as is or with more info)